web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsWriter.java

```diff
@@ -225,236 +313,284 @@ public void run() {
                     log.error(e.getMessage(), e);
                 }
             }
+            if (!failedMetrics.isEmpty()) {
+                writeFailedMetrics(failedMetrics);
+            }
         }
-    }
 
-    private void processQueryMetrics(List<QueryMetricHolder> metricHolderList) {
-        if (!metricHolderList.isEmpty()) {
-            if (writerConfig.getUseRemoteService()) {
-                processQueryMetricsWithRemoteService(metricHolderList);
-            } else {
-                processQueryMetricsWithHandler(metricHolderList);
+        /**
+         * Process query metric updates using either the RemoteQueryMetricService or the ShardTableQueryMetricHandler
+         *
+         * @param metricHolderList
+         */
+        private void processQueryMetrics(List<QueryMetricHolder> metricHolderList) {
+            if (!metricHolderList.isEmpty()) {
+                if (writerConfig.getUseRemoteService()) {
+                    processQueryMetricsWithRemoteService(metricHolderList);
+                } else {
+                    processQueryMetricsWithHandler(metricHolderList);
+                }
             }
         }
-    }
 
-    private void processQueryMetricsWithRemoteService(List<QueryMetricHolder> metricHolderList) {
-        List<BaseQueryMetric> metricList = metricHolderList.stream().map(QueryMetricHolder::getQueryMetric).collect(Collectors.toList());
-        if (!metricList.isEmpty()) {
+        /**
+         * Process query metric updates using the RemoteQueryMetricService
+         *
+         * @param metricHolderList
+         */
+        private void processQueryMetricsWithRemoteService(List<QueryMetricHolder> metricHolderList) {
+            List<BaseQueryMetric> metricList = metricHolderList.stream().map(QueryMetricHolder::getQueryMetric).collect(Collectors.toList());
+            if (!metricList.isEmpty()) {
+                try {
+                    writeMetricsToRemoteService(metricList);
+                    log.debug(String.format("wrote %d metric updates to RemoteQueryMetricService", metricList.size()));
+                } catch (Exception e) {
+                    log.error(String.format("%d metric updates failed write to RemoteQueryMetricService as a batch, will retry individually - %s",
+                                    metricList.size(), e.getMessage()), e);
+                    metricList.forEach(m -> {
+                        failedMetrics.add(new FailureRecord(m, false));
```

Comment:
Why have failed metrics at all?  Why not just put the failed metric back into the blocking queue?

Reply:
To track whether the failure is because of the metric itself or a system failure.  The data stored in FailureRecord helps weather and recover from metric-based, intermittent, and long-running failures

Reply:
Also -- the primary reason for this PR is to have multiple MetricProcessor threads reading off of the queue for the remote service case.  The other functionality is unchanged.